src/fileops.rs

```diff
@@ -7,21 +7,52 @@ extern crate libc;
 
 use std::ffi::CString;
 use libc::c_int;
+use std::os::unix::io::{RawFd, AsRawFd};
+use std::ops::Drop;
 
-pub fn open<T: Into<String>>(path: T, mode: c_int) -> c_int {
-  unsafe { libc::open(CString::new(path.into())
-                      .unwrap_or(CString::new("").unwrap()).as_ptr(), mode) }
+pub struct Fd(RawFd);
+
+impl Fd {
+  pub fn open<T: Into<String>>(path: T, mode: c_int) -> Option<Fd> {
+    let fd = unsafe { libc::open(CString::new(path.into())
+                .unwrap_or(CString::new("").unwrap()).as_ptr(), mode) };
+    match fd {
+      -1 => None,
+      _ => Some(Fd(fd))
+    }
+  }
+
+  fn close(&self) -> c_int {
+    unsafe { libc::close(self.0) }
+  }
+
+  pub fn fcntl(&self, cmd: c_int, arg: c_int ) -> Option<c_int> {
+    let ret = unsafe { libc::fcntl(self.0, cmd, arg) };
+    match ret {
```

Comment:
Optional: You can just do this if you'd like I think, since "ret" is a pretty unhelpful name

```rust
match unsafe { libc::fcntl(...) } {
}
```